libraries/chain/steem_evaluator.cpp

```diff
@@ -713,11 +724,13 @@ namespace golos { namespace chain {
                     const auto &comment = *itr;
                     if ( !_db.has_hardfork( STEEMIT_HARDFORK_0_18__536 ) ) {
                         if (_db.has_hardfork(STEEMIT_HARDFORK_0_14__306)) {
-                            FC_ASSERT(_db.calculate_discussion_payout_time(comment) != fc::time_point_sec::maximum(),
-                                      "The comment is archived.");
+                            GOLOS_CHECK_LOGIC(_db.calculate_discussion_payout_time(comment) != fc::time_point_sec::maximum(),
```

Comment:
Will never get here too (whole `if ( !_db.has_hardfork( STEEMIT_HARDFORK_0_18__536 ) ) {…}`)

Reply:
I agree, this check can be removed too.